selftests/functional/basic.py

```diff
@@ -98,6 +98,12 @@ def test(self):
         logging.getLogger("some.other.logger").info("SHOULD BE ON debug.log")
 """
 
+EXEC_ENV_VARIABLE_TEST = """#!/bin/bash
+if [[ -n "${TEST_ENV}"  ]]; then
```

Comment:
This of course will arguably-incorrectly pass if `$TEST_ENV` is present but an empty string.  That's hard to fix in shell though, so it's probably acceptable.

Reply:
I agree, but I think that for testing purposes is fine.